desktop/src/main/java/bisq/desktop/components/paymentmethods/GeneralBankForm.java

```diff
@@ -136,26 +138,61 @@ void updateHolderIDInput(String countryCode, boolean requiresHolderId) {
     }
 
     void autoFillAccountTextFields(CountryBasedPaymentAccountPayload paymentAccountPayload) {
+
+
+        BankAccountPayload bankAccountPayload = null;
+        CashDepositAccountPayload cashDepositAccountPayload = null;
+        if (paymentAccountPayload instanceof BankAccountPayload) {
+            bankAccountPayload = (BankAccountPayload) paymentAccountPayload;
+        } else if (paymentAccountPayload instanceof CashDepositAccountPayload) {
+            cashDepositAccountPayload = (CashDepositAccountPayload) paymentAccountPayload;
+        }
+
```

Comment:
I don't like it so much having those checks in the abstract class for specific implementations. I think it would be better to have a getBankId, getBranchId, getBankName and getAccountNr implementation in the CashDeposit and BankAccount forms WDYT?

Reply:
Fixed in latest commit.

Code could be even more simplified when `CashDepositAccountPayload` extends `BankAccountPayload` - do you see any obstacles from doing it (in this pull request)?

Reply:
I think - let's keep it to a minimum for now. I'll test it now.